src/main/java/com/puppycrawl/tools/checkstyle/checks/TrailingCommentCheck.java

```diff
@@ -317,7 +317,7 @@ private boolean isLegalSingleLineComment(String comment) {
         }
         else {
             // remove chars which start comment
-            final String commentText = comment.substring(1).trim();
+            final String commentText = comment.trim();
```

Comment:
Please remove comment above.

Why we trim comment ? In general, if our Check is regexp based

Reply:
Agreed, done

Reply:
As a decision for new features, cool. As a fix to a regression, I'd prefer to restore the old behavior (trimmed comment matched against regexp).

Reply:
@hd42 check behavior has not changed; we only check that given regex exists in comment. There was no need to trim or to use substring.